core/modules/ccontrol/MergingHandler.h

```diff
@@ -45,6 +46,39 @@ namespace lsst {
 namespace qserv {
 namespace ccontrol {
 
+/// A class to delay creating the buffer until it is requested
+/// by xrootd SSI. The size of the buffer that will be needed is
+/// set using setTargetSize(int sz), and the buffer of that size is
+/// created by calling resizeToTargetSize(). When a buffer is no longer
+/// needed, zero() should be called to free the memory.
+class MergeBuffer {
+public:
+    using bufType = std::vector<char>;
+    MergeBuffer() {
+        _id = "MBI=" + std::to_string(_sequence++);
+        zero();
+    }
+    virtual ~MergeBuffer();
+
+    std::shared_ptr<bufType> getBuffer();
+    size_t getSize();
+    size_t getTargetSize() { return _targetSize; }
+    void setTargetSize(int sz);
+    void resizeToTargetSize();
+    void zero(); ///< Set buffer size and _targetSize to zero, ensure memory is freed.
```

Comment:
The method should have a bit more meaningful name like:

 void resetTargetSize()

Or even better, you may simplify things even further by eliminating that method altogether and adding the default value to the following method instead:

  void setTargetSize(int sz=0);